firefox-ios/firefox-ios-tests/Tests/XCUITests/CookiePersistenceTests.swift

```diff
@@ -0,0 +1,96 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import XCTest
+
+final class CookiePersistenceTests: BaseTestCase {
+    let cookieSiteURL = "http://localhost:\(serverPort)/test-fixture/test-cookie-store.html"
+    let websites = ["http://mozilla.org",
+                    "http://wikipedia",
+                    "http://reddit.com"]
+
+    override func setUp() {
+        // Fresh install the app
+        // removeApp() does not work on iOS 15 and 16 intermittently
+        if #available(iOS 17, *) {
+            removeApp()
+        }
+
+        // The app is correctly installed
+        super.setUp()
+    }
+
+    func testCookiePersistenceBasic() {
+        // Open URL for Cookie login
+        openCookieSite()
+        let webview = app.webViews.firstMatch
+        mozWaitForElementToExist(webview.staticTexts["Cookie Test Page"])
+        mozWaitForElementToExist(webview.staticTexts["LOGGED_OUT"])
+
+        // Tap on Log in
+        webview.buttons["Login"].waitAndTap()
+
+        mozWaitForElementToExist(webview.staticTexts["LOGGED_IN"])
+
+        // Relaunch app
+        relaunchApp()
+
+        openCookieSite()
+        mozWaitForElementToExist(webview.staticTexts["Cookie Test Page"])
+        mozWaitForElementToExist(webview.staticTexts["LOGGED_IN"])
+    }
+
+    func testCookiePersistence_OpenNewTab() {
+        // Open URL for Cookie login
+        openCookieSite()
+        let webview = app.webViews.firstMatch
+        mozWaitForElementToExist(webview.staticTexts["Cookie Test Page"])
+        mozWaitForElementToExist(webview.staticTexts["LOGGED_OUT"])
+
+        // Tap on Log in
+        webview.buttons["Login"].waitAndTap()
+
+        mozWaitForElementToExist(webview.staticTexts["LOGGED_IN"])
+
+        // Open a few tabs
+        websites.forEach { website in
```

Comment:
@isabelrios In this step I'm trying to open a few tabs to go through the process of creating a tab, selecting etc, which has caused cookie issues in the past, but the test is not passing locally because it is not typing the address I'm not sure what I'm missing